src/molecule/command/init/role.py

```diff
@@ -58,8 +59,22 @@ def execute(self):
 
         :return: None
         """
+        namespace = None
         role_name = self._command_args["role_name"]
         role_directory = os.getcwd()
+
+        # outside collections our tooling needs a namespace.
+        if not os.path.isfile("../galaxy.yml"):
+            name_re = re.compile(r"^[a-z][a-z0-9_]+\.[a-z][a-z0-9_]+$")
+
+            if not name_re.match(role_name):
```

Comment:
I will merge that like this because fixing that bug is quite important. My plan was to add code to ansible-compat for doing this but that may take extra time. I will look into replacing this code with the one from compat, once we have it.